support/modules-artifact-gen/directory-artifact-gen

```diff
@@ -161,23 +161,46 @@ else
   exit 1
 fi
 
-
 # Create dest_dir file in plain text
 echo "$dest_dir" > $dest_dir_file
 
+# Check the the passthrough_args and potentially modify them
+# to avoid conflicts or to let them override the already args
+# provided to mender-artifact
+passthrough_args_modified=" "
+echo $passthrough_args | xargs -n 2 printf "%s %s\n" | (while read -r flag arg; do
+  if [ -n "$flag" ] && [ -n "$arg" ]; then
+    case $flag in
+      -T | --type)
+        echo "Error: Conflicting flag '$flag'. Already specified by the script."
+        exit 1
+        ;;
+      -o | --output-path)
+        output_path=$arg
+        ;;
+      -n | --name)
+        artifact_name=$arg
+        ;;
+      *)
+        passthrough_args_modified="$passthrough_args_modified $flag $arg"
+        ;;
+    esac
+  fi
+done
+
 mender-artifact write module-image \
   -T directory \
   $device_types \
   -o $output_path \
   -n $artifact_name \
   -f $update_files_tar \
   -f $dest_dir_file \
-  $passthrough_args
+  $passthrough_args_modified
 
 rm $update_files_tar
 rm $dest_dir_file
 
 echo "Artifact $output_path generated successfully:"
 mender-artifact read $output_path
-
+)
```

Comment:
Good catch with the subshell. I didn't see this coming! :exploding_head: 

To make it more obvious I recommend 1) adding a good comment on why we need the subshell (to be able to override the variables then passed to `mender-artifact` commands) and 2) moving the `rm` commands out of the subshell, as these variables are not being overridden.